### ISSUE TITLE
ABIv2Decoder fails to decode empty arrays

### ISSUE THREAD

**Issue by wingsofovnia** [BANKEX/web3swift#120]
## Contract
```
function empty() external view returns(uint[]) {
        uint[] memory result = new uint[](0);
        return result;
}
```
## Calling code
```
let txIntermediate = self.contract.method("empty", parameters: [AnyObject](), options: self.web3opt)
let txResponse = txIntermediate?.call(options: self.web3opt)
```
## Error
**What:** Fatal error: Can't form Range with upperBound < lowerBound
**Where:** `ABIv2Decoder#decodeSignleType` -> `case .array:` -> `dataSlice = elementItself[32 ..< subType.memoryUsage*length]` https://github.com/BANKEX/web3swift/blob/3d06c158ad1a94f5a5588636ce7faf01885320f9/web3swift/ABIv2/Classes/ABIv2Decoding.swift#L113
**Reason:** length is 0 ->  upperBound = 0?

**Comment by shamatar** [comment]
Hello @wingsofovnia 

Thank you for a big report, will fix it tomorrow. 

Sincerely, Alex

**Comment by shamatar** [comment]
Hello @wingsofovnia 

Should be fixed in #123 Please reopen in problem persists.

Sincerely, Alex
